cli/flags.rs

```diff
@@ -373,8 +384,14 @@ mod tests {
 
   #[test]
   fn test_flags_from_vec_6() {
-    let (flags, subcommand, argv) =
-      flags_from_vec(svec!["deno", "--allow-net", "gist.ts", "--title", "X"]);
+    let (flags, subcommand, argv) = flags_from_vec(svec![
+      "deno",
+      "--allow-net",
+      "run",
```

Comment:
It seems `--allow-net` should come after `run` because it's an argument of run, not of deno itself. WDYT?

Reply:
Yep, we discussed it in #2228. I'll add it to this PR

Reply:
Done, I've took the liberty and made remaining flags global.